### PR TITLE
feat(alerts): Introduce RuleSource for the Rule model

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This will be used as a way to differentiate the purpose of alert rules.

Specifically for crons we need a way to distinguish between rules created for cron monitors so that we can keep them hidden from the user in the alerts rule list.

### CODE REVIEW

File: src/sentry/models/rule.py [hunk 1]
```diff
@@ -42,6 +55,13 @@ class Rule(Model):
         choices=((RuleStatus.ACTIVE, "Active"), (RuleStatus.INACTIVE, "Inactive")),
         db_index=True,
     )
+    # type is currently used as a way to distinguish rules created specificly
+    # for use in other parts of the product (e.g. monitor alerting rules)
+    type = BoundedPositiveIntegerField(
+        default=RuleType.ISSUE,
+        choices=RuleType.as_choices(),
+        db_index=True,
```
Comment (wedamija) [inline src/sentry/models/rule.py:63]: I think you'd be better off with an index on `project, type`. We're rarely going to be just filtering on type

Comment (evanpurkhiser) [reply src/sentry/models/rule.py:63]: good point

File: src/sentry/models/rule.py [hunk 2]
```diff
@@ -24,6 +25,18 @@ class RuleStatus:
     DELETION_IN_PROGRESS = 3
 
 
+class RuleType(IntEnum):
+    ISSUE = 0
+    CRON_MONITOR_RULE = 1
+
+    @classmethod
+    def as_choices(cls) -> Sequence[Tuple[int, str]]:
```
Comment (JoshFerge) [inline src/sentry/models/rule.py:33]: aren't metric alerts another rule type? or is that a whole separate thing

Comment (evanpurkhiser) [reply src/sentry/models/rule.py:33]: lol told you this would be confusing @wedamija

Yeah they are a totally different thing

Comment (JoshFerge) [reply src/sentry/models/rule.py:33]: add a comment? :D

Comment (wedamija) [reply src/sentry/models/rule.py:33]: lol, we think we've come up with a clearer name

**Review by wedamija — APPROVED** [review]
lgtm. We might need to split the index into a separate pr, since there are 13mm rows in this table and creation might time out.

**Comment by evanpurkhiser** [comment]
Index split

**Pull request merged**